src/coreclr/jit/lower.cpp

```diff
@@ -3265,6 +3266,26 @@ GenTree* Lowering::OptimizeConstCompare(GenTree* cmp)
 #endif // TARGET_XARCH
 #endif // defined(TARGET_XARCH) || defined(TARGET_ARM64)
 
+    // Optimize EQ/NE(relop/SETCC, 0) into (maybe reversed) cond.
+    if (op2->IsIntegralConst(0) && (op1->OperIsCompare() || op1->OperIs(GT_SETCC)))
+    {
+        LIR::Use use;
+        if (BlockRange().TryGetUse(cmp, &use))
+        {
+            if (cmp->OperIs(GT_EQ))
+            {
+                GenTree* reversed = comp->gtReverseCond(op1);
+                assert(reversed == op1);
+            }
+
+            GenTree* next = cmp->gtNext;
+            use.ReplaceWith(op1);
+            BlockRange().Remove(cmp->gtGetOp2());
+            BlockRange().Remove(cmp);
+            return next;
```

Comment:
The difficulty I have with debugging the lowering phase, is that nodes can get renamed or moved, and there is no record in the JITDUMP except for the node dump before and after the phase. It's not always obvious what happened, especially when your work isn't focused on the lowering phase.
Not for this PR, but some extra debug throughout lowering would be useful.

Reply:
I agree, we should have more dump logging in lowering. I'll add some to this transform.

Reply:
The additional dumping looks good. I think the "before" IR isn't required in these cases as that IR would have been printed the last time it was changed, but happy for it to stay like this.